azure/create-spn.sh

```diff
@@ -135,24 +150,70 @@ show_account()
 
 create_ad_sp()
 {
-  echo "Creating Azure AD Service Principal and configuring its access to Azure resources in subscription ${subscription}..."
-  years="${years:-1}"
-  spn=$(az ad sp create-for-rbac -o json -n ${display_name} --role Owner --scopes /subscriptions/${subscription} --years ${years})
+  if [[ -z "${client_id}" ]]; then
+    echo "Creating Azure AD Service Principal and configuring its access to Azure resources in subscription ${subscription}..."
+    years="${years:-1}"
+    spn=$(az ad sp create-for-rbac -o json -n "${display_name}" --only-show-errors \
+      --role Owner --scopes /subscriptions/${subscription} --years ${years})
```

Comment:
we're inconsistent with the quotes, 
we should prefer "${subscription}" instead of ${subscription}

Reply:
Updated